### ISSUE TITLE
IAMServiceAccount project is "default", causes 403

### ISSUE THREAD

**Issue by kevinohara80** [GoogleCloudPlatform/k8s-config-connector#86]
According to the workload identity docs, once config connector is deployed and tested, IAM service accounts should be configured like this

```yaml
apiVersion: iam.cnrm.cloud.google.com/v1beta1
kind: IAMServiceAccount
metadata:
  name: my-service-account
spec:
  displayName: my-service-account
```

Things weren't working so I pulled the IAMServiceAccount from Kubernetes. I noticed that there was a `cnrm.cloud.google.com/project-id` annotation added but it was set to `default`. Looking at the `status` in the manifest, you can clearly see that the error has to do with that property. Here's the pulled manifest...

```yaml
apiVersion: iam.cnrm.cloud.google.com/v1beta1
kind: IAMServiceAccount
metadata:
  annotations:
    cnrm.cloud.google.com/project-id: default
  creationTimestamp: "2020-02-12T18:40:31Z"
  generation: 2
  labels:
    app.kubernetes.io/instance: my-service
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: my-service
    app.kubernetes.io/version: 1.16.0
    helm.sh/chart: my-service-0.1.0
  name: my-service-account
  namespace: default
  resourceVersion: "278101"
  selfLink: /apis/iam.cnrm.cloud.google.com/v1beta1/namespaces/default/iamserviceaccounts/my-service
  uid: 25728880-4dc7-11ea-a07c-42010a800133
spec:
  displayName: my-service-account
status:
  conditions:
  - lastTransitionTime: "2020-02-12T18:40:31Z"
    message: 'Update call failed: error fetching live state: error reading underlying
      resource: Error reading Service Account "projects/default/serviceAccounts/my-service@default.iam.gserviceaccount.com":
      googleapi: Error 403: Permission iam.serviceAccounts.get is required to perform
      this operation on service account projects/default/serviceAccounts/my-service@default.iam.gserviceaccount.com.,
      forbidden'
    reason: UpdateFailed
    status: "False"
    type: Ready
```

**I actually fixed this by hardcoding the annotation to the correct project ID** and everything works. So the question is, are you required to set the `cnrm.cloud.google.com/project-id` or is there somewhere else that `project_id` needs to be provided?

**Comment by austbot** [comment]
@kevinohara80 I can confirm this worked for me too. I don't know who to tell to update docs on the GCP documentation. Admins I hope you are listening ! :)

**Comment by caieo** [comment]
Hi @kevinohara80 & @austbot , sorry you ran into this, I will forward this to the doc team to emphasize some steps. 
The annotation for `project-id` pulls 'default' from the namespace, so the ["Setting your default namespace"](https://cloud.google.com/config-connector/docs/how-to/install-upgrade-uninstall#setting_your_default_namespace) step after installation in actually more important than we make it seem. Hardcoding the annotation to the correct project ID works, but would be a hassle to keep doing if the project-id annotation keeps getting filled incorrectly.

**Comment by austbot** [comment]
@caieo Hey thanks for responding. 
I'm not sure the namespace has anything to do with this issue. As the ConfigConnector is installed in the appropriate namespace. Perhaps I'm mistaken, but do you mean set the default project in the ConfigConnector configuration :) ?

**Comment by caieo** [comment]
@austbot Some context on how ConfigConnector will use namespaces: ConfigConnector will first check if the resource has an explicit annotation; if that doesn't work, it will check if the namespace has an annotation; and if the namespaced annotation doesn't exist, the final resort is to pull the namespace name (which in this case ended up as 'default'). 

In addition to the ["setting your default namespace"](https://cloud.google.com/config-connector/docs/how-to/install-upgrade-uninstall#setting_your_default_namespace) you can also refer to ["Annotate your namespace with a default project ID"](https://cloud.google.com/config-connector/docs/how-to/setting-default-namespace#annotate_your_namespace_with_a_default_project_id) to annotate your namespace. 

I will make sure to get the namespace/project-id correlation emphasized more in our docs, but does this explanation clarify the situation?

**Comment by austbot** [comment]
@caieo Thank you for explaining, yes it resolves the issue. I wasn't aware of how config connector looks at annotations on the namespace. Its very possible that I missed it in the docs.

**Comment by caieo** [comment]
@austbot  The fact that both you and @kevinohara80  ran into this issue means that we should improve the docs for installation setup, so thank you again for bringing it up & I'm happy we were able to resolve the issue!

**Comment by kevinohara80** [comment]
@caieo Thank you! This works now.

P.S. Setting the default namespace for kubectl doesn't really address the issue at all. That step in the installation guide is just meant to show you that, for convenience, you can configure kubectl to use a namespace other than default. The missing piece for Config Connector to work properly was the annotation on the namespace in k8s which isn't mentioned in the installation guide which is why we both missed it.

**Comment by dinvlad** [comment]
There's further nuance here (and possibly a bug in the error message?). I had exactly the same error as @kevinohara80, and I've already set `project-id` annotation both in the default and my custom namespaces. Still, this wasn't working.

However, my CNRM service account had a custom role with `iam.serviceAccounts.get` and `iam.serviceAccounts.create` permissions, so the error message `Permission iam.serviceAccounts.get is required` didn't really make sense. Then I realized that Config Connector probably does a "list" operation on the service accounts in the projects, instead of "get" on a specific one (as would be implied from the message). Indeed, once I added `iam.serviceAccounts.list` permission to the role, this finally worked.
